File: current_session/101.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return helper(root.left, root.right);
    }

    private boolean helper(TreeNode l, TreeNode r) {
        if (l == null && r != null) return false;
        if (r == null && l != null) return false;
        if (r == null && l == null) return true;
        if (r.val != l.val) return false;
        return helper(r.right, l.left) && helper(l.right, r.left);
    }
}

File: current_session/101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def helper(l, r):
            if not r and l: return False
            if not l and r: return False
            if not l and not r: return True
            if r.val != l.val: return False
            return helper(r.left, l.right) and helper(r.right, l.left)
        return helper(root.left, root.right)